core/libretro/libretro.cpp

```diff
@@ -578,6 +586,24 @@ static void update_variables(bool first_startup)
    int i ;
    char key[256] ;
 
+   var.key = "reicast_per_content_vmus";
+   unsigned previous_per_content_vmus = per_content_vmus;
+   per_content_vmus = 0;
+
+   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
+   {
+      if (!strcmp("VMU A1", var.value))
+         per_content_vmus = 1;
+      else if (!strcmp("All VMUs", var.value))
+         per_content_vmus = 2;
+   }
+
+   if (per_content_vmus != previous_per_content_vmus)
+   {
+      mcfg_DestroyDevices();
+      mcfg_CreateDevices();
```

Comment:
This will cause problems with Naomi: the maple devices should not be changed after the game has started.

In this case adding `&& settings.System == DC_PLATFORM_DREAMCAST` to the if condition should do the trick.

Reply:
Thanks - I've made the change you suggested